packages/backend/jobs-manager/service/src/modules/database/subscriptions/cron-subscriptions/cron-subscriptions.service.ts

```diff
@@ -66,6 +67,17 @@ export class CronSubscriptionsService {
     return await this.subscriptionModel.create(sub);
   }
 
+  public async updateEnabled(id: string, enabled: boolean) {
+    const subUpdate: Partial<CronSubscription> = {
+      isEnabled: enabled,
+    };
+
+    return await this.subscriptionModel.updateOne<CronSubscription>(
+      { _id: { $eq: new Types.ObjectId(id) } },
+      subUpdate,
+    );
+  }
+
   public async getAll() {
     return await this.subscriptionModel.find({}, '-file');
   }
```

Comment:
I can't add a comment down there because it is not a change, but basically, here is the end of my previous comment about disabling cron subscriptions. We should do the check in the jobs manager service as well as a fail safe. Something like the following:

```typescript
 public async launchCronSubscriptionJob(id: string) {
    const sub: CronSubscriptionsDocument =
      await this.subscriptionModel.findById(id);
    if (!sub) {
      this.logger.warn(
        `Cron subscription id '${id}' does not exist. Could not launch a job from non-existant cron subscription`,
      );
      return;
    }

    if (!sub.isEnabled) return; // soft fail here as it is likely a cache delay issue

    sub.jobParameters =
      await SubscriptionsUtils.getParametersForCustomJobSubscription(
        sub,
        this.logger,
        this.customJobsService,
        this.configService,
      );
    if (!sub.jobParameters) return;
    await this.setupSubscriptionsForProjects(sub);
  }
```

The previous code is a slight modification of the already existing `launchCronSubscriptionJob` method. Maybe it should be the following if `isEnabled` being `null` means that it is, in fact, enabled.

```typescript
    if (sub.isEnabled === false) return; 
```

It would also be possible to simply only get enabled subscriptions, if more convenient.

Reply:
Initially, I only put the disable check in one spot because I thought the code might be confusing if we have multiple places in the code doing the check, but it's a really good point about the delay. 👍🏻